core/cache/model.go

```diff
@@ -245,12 +248,19 @@ func (m *Model) updateUnit(ch UnitChange) {
 	m.mu.Unlock()
 }
 
+// unitLXDProfileRemove contains an appName and it's charm URL.  To be used
+// when publishing for modelUnitLXDProfileRemove.
+type unitLXDProfileRemove struct {
+	name    string
+	appName string
+}
+
 // removeUnit removes the unit from the model.
 func (m *Model) removeUnit(ch RemoveUnit) {
 	m.mu.Lock()
 	unit, ok := m.units[ch.Name]
 	if ok {
-		m.hub.Publish(m.topic(modelUnitLXDProfileChange), []string{ch.Name, unit.details.Application})
+		m.hub.Publish(m.topic(modelUnitLXDProfileRemove), unitLXDProfileRemove{name: ch.Name, appName: unit.details.Application})
```

Comment:
I was going back and forth on the same.  Since the MachineAppLXDProfileWatcher needs more data on unit removal than simply knowing what unit it is, I decided to be more specific.  Though there is no reason this can't be renamed for more generic use, as long as the contents of unitLXDProfileRemove{} don't decrease.